spec/std/signal_spec.cr

```diff
@@ -36,6 +36,24 @@ describe "Signal" do
       Process.signal Signal::USR2, Process.pid
     end
 
+    it "allows chaining of signals" do
+      ran_first = false
+      ran_second = false
+
+      Signal::USR1.trap { ran_first = true }
+      existing = Signal::USR1.trap_handler?
+
+      Signal::USR1.trap do |signal|
+        existing.try &.call(signal)
+        ran_second = true
+      end
+
+      Process.signal Signal::USR1, Process.pid
+      sleep 0.1
+      ran_first.should be_true
+      ran_second.should be_true
+    end
```

Comment:
To avoid test contamination with other tests, should this reset the USR1 trap (similar to line 64/65 below)?  I'm actually surprised most of the other tests don't reset any of the signals they are trapping.

Reply:
The spec runner is not expected to typically receive any signals (except for termination). So there's no actual effect from this. But resetting is better, of course.

Reply:
should I add at the end of the spec?

```crystal
ensure
      Signal::USR1.reset
```

Reply:
That's what I was thinking, yes.